runelite-client/src/main/java/net/runelite/client/plugins/party/PartyPlugin.java

```diff
@@ -141,17 +156,35 @@ public void configure(Binder binder)
 	@Override
 	protected void startUp() throws Exception
 	{
+		panel = new PartyPanel(this, config, party);
```

Comment:
Usually if we want to build a panel that contains a bunch of injectable objects, we instead manually pull it from the injector during startUp(), which correctly will construct it on the swing thread. Eg `injector.getIntance()`. Search some of the other plugins which add panels.